Docs: Plugin-E2E: Docs improvements (#778)
grafana/plugin-tools: docusaurus/docs/e2e-test-a-plugin/get-started.md

import ScaffoldPluginE2EDSWorkflowYarn from '@snippets/plugin-e2e-ds-workflow.yarn.md';
import ScaffoldPluginE2EDSWorkflowPNPM from '@snippets/plugin-e2e-ds-workflow.pnpm.md';

> This guide will walk through how to get started with end-to-end testing in your plugin.

Reviewer: ```suggestion
This guide walks you through how to get started with end-to-end testing in your plugin.
```

> **You will learn:**

Reviewer: ```suggestion
**Topics include:**
```

> - how to install `@grafana/plugin-e2e`

Reviewer: ```suggestion
- How to install `@grafana/plugin-e2e`
```

> - how to configure authentication

Reviewer: ```suggestion
- How to configure authentication
```

> - how to write tests

Reviewer: ```suggestion
- How to write tests
```

> - how to setup a basic Github workflow that runs tests against multiple versions of Grafana

Reviewer: ```suggestion
- How to setup a basic Github workflow that runs tests against multiple versions of Grafana
```

### Installing Playwright

> `@grafana/plugin-e2e` extends Playwright APIs, so you need to have `@playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the package.json file of your plugin. Please refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install. Make sure you can run the example tests that were generated when you installed Playwright before you proceed to the next step in this guide.

Reviewer: ```suggestion
The `@grafana/plugin-e2e` tool extends Playwright APIs, so you need to have `@playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the `package.json` file of your plugin. Refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install Playwright. Make sure you can run the example tests that were generated when you installed Playwright before you proceed to the next step in this guide.
```

> ## Set up @grafana/plugin-e2e

Reviewer: ```suggestion
## Set up `@grafana/plugin-e2e`
```

Reviewer: See comment in GitHub conversations regarding this. Grafana style, following Google, prefers `code formatting` in headers so long as a code term is not used alone.

});
```

> The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`. To prevent these files from being version controlled, you can add the following line to your `.gitignore` file.

Reviewer: ```suggestion
The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`. To prevent these files from being version controlled, you can add the following line to your `.gitignore` file:
```

### Step 6: Run tests in CI

We recommend using a CI workflow to run end-to-end tests to continuously check for breakages. The following Github workflow will run end-to-end tests against a range of Grafana versions for every PR in your Github repository.

:::note

> This is a generic example based on a backend plugin. You may need to alter or remove some of the steps in the `playwright-tests` job before using it in your plugin.

Reviewer: ```suggestion

This is a generic example based on a backend plugin. You may need to alter or remove some of the steps in the `playwright-tests` job before using it in your plugin.

```
